pkg/daemon/drain.go

```diff
@@ -223,12 +223,20 @@ func isSafeContainerRegistryConfChanges(oldConfig, newConfig *mcfgv1.MachineConf
 
 	oldRegHashMap := make(map[string]sysregistriesv2.Registry)
 	for _, reg := range tomlConfOldReg.Registries {
-		oldRegHashMap[reg.Location] = reg
+		scope := reg.Location
+		if reg.Prefix != "" {
+			scope = reg.Prefix
+		}
+		oldRegHashMap[scope] = reg
 	}
 
 	newRegHashMap := make(map[string]sysregistriesv2.Registry)
 	for _, reg := range tomlConfNewReg.Registries {
-		newRegHashMap[reg.Location] = reg
+		scope := reg.Location
+		if reg.Prefix != "" {
+			scope = reg.Prefix
+		}
+		newRegHashMap[scope] = reg
```

Comment:
A bit below, I think the `reflect.DeepEqual` should also verify `Location` is unchanged (that used to be implied by using `Location` as the map key, but isn’t anymore).

Reply:
fixed